sigmastate/src/main/scala/sigmastate/AvlTreeData.scala

```diff
@@ -92,8 +92,10 @@ object AvlTreeData {
     override def parse(r: SigmaByteReader): AvlTreeData = {
       val digest = r.getBytes(DigestSize)
       val tf = AvlTreeFlags(r.getByte())
-      val keyLength = r.getUIntExact
-      val valueLengthOpt = r.getOption(r.getUIntExact)
+      val keyLength = r.getUInt().toInt
+      val valueLengthOpt = r.getOption(r.getUInt().toInt)
+      // Note, when keyLength and valueLengthOpt < 0 as a result of Int overflow,
+      // the deserializer succeeds with invalid AvlTreeData
```

Comment:
Will invalid AvlTreeData fail validation after? If so , why not to keep  `getUIntExact` ?

Reply:
Some AvlTree operations (remove_eval, update_eval, contains_eval) won't throw. Added comment in this code and also added test.